content/tutorials/trainapp-fitbit.textile

```diff
@@ -0,0 +1,531 @@
+    const towardsOptions = viaOptions[0].towards.map(mapping => {
+      return { name: mapping };
+    });
+    // Update the via with the new value 
+    settingsStorage.setItem(
+      "via",
+      `{"values":${JSON.stringify(towardsOptions)}} `
+    );
+  } else if (evt.key == "towards") {
+    // We get the old and new value of the setting
+    const towardsOld = JSON.parse(evt.oldValue).name;
+    const towardsNew = JSON.parse(evt.newValue).name;
+    // If same we just return 
+    if (towardsOld == towardsNew) return;
+  }
+  sendMessage();
+};
+// Listen for the onerror event
+messaging.peerSocket.onerror = function(err) {
+  // Handle any errors
+  console.log("Connection error: " + err.code + " - " + err.message);
+};
+```
+
+"See this step in Github":https://github.com/ably/tutorials/commit/55d2b927547b6b7520fe4bde2dec0fa0948ae152
+
+So this completes the work of the companion, now we need to move to the app (which is what is displayed on the watch).
+
+h2(#send-message). Step 8 - Update the gui files of app
+
+Now we update @gui@ files of the app to get the look of the clockface and see actual values on the watch.
+
+Empty the file @index.gui@ which is inside the @resources@ folder and add the following code: 
+
+```[gui]
+<svg viewport-fill="black">
+<!-- Clock -->
+  <text id="myClock" x="50%" y="40%" font-size="32" font-family="System-Regular" 
+  text-anchor="middle" text-length="20" fill="fb-cyan">00:00:00</text>
+  <!-- Train time -->
+  <text id="trainlabel" x="3%" y="10%" fill="fb-indigo" font-family="Seville-Bold-Italic" 
+  text-length="30"></text>
+  <text id="teta" x="6%" y="20%" fill="fb-lavender" text-length="30"></text>
+  <!-- Walk time -->
+  <text id="walkinglabel" x="54%" y="89%" fill="fb-indigo" font-family="Seville-Bold-Italic" 
+  text-length="30"></text>
+  <text id="weta" x="66%" y="98%" fill="fb-lavender" text-length="30"></text>
+  <!-- Spinner -->
+  <use id="spinner" href="#spinner" x="50%-25" y= "50%-25" 
+  width="50" height="50" fill="fb-lavender" />
+  <!-- Arc -->
+  <arc id="cir" x="50" y="50" width="200" height="190" 
+  fill="fb-green" arc-width="6" start-angle="0" sweep-angle="360">
+    <!-- Leave in -->
+    <text id="leavelabel" x="33%" y="55%" fill="fb-indigo" 
+    font-family="Seville-Bold-Italic" text-length="30"></text>
+    <text id="eta" x="38%" y="65%" fill="fb-green" 
+    font-family="Fabrikat-Bold" text-length="30"></text>
+  </arc>
+</svg>
+```
+
+Update the file @widgets.gui@ which is inside the @resources@ folder with the following code:
+
+```[gui]
+<svg>
+  <defs>
+    <link rel="stylesheet" href="styles.css" />
+    <link rel="import" href="/mnt/sysassets/widgets_common.gui" />
+    <link rel="import" href="/mnt/sysassets/widgets/spinner_widget.gui" />
+  </defs>
+</svg>
+```
+
+Empty the file @styles.css@ which is inside the @resources@ folder. We have added the styles in the gui files and we are also updating them in the @index.js@ of the app depending on the values. 
```

Comment:
Delete the styles.css file, which is inside the resources folder and remove the line which links it